### PR TITLE
ci: add vpm workflow

### PR DESCRIPTION
**Pull request by ttytm**
Due to the recent addition of the requirement to add `-d network` to include VPM tests into test runs, there is no CI coverage for VPM. The tests are always skipped because there are no runners defining this flag.

This PR adds a VPM workflow. It only runs when VPM related files or the workflow file have been changed, so there is no additional overhead when other changes are made in the codebase.

https://github.com/ttytm/v/actions/runs/7212169683

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 35077c7</samp>

Add a GitHub Actions workflow to run CI tests for `vpm` on different OS and C compilers. The workflow file is `.github/workflows/vpm_ci.yml`.

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 35077c7</samp>

* Add a new workflow file for GitHub Actions to run CI tests for vpm (F0)

### CODE REVIEW

**Comment by spytheman** [comment]
Thanks, I was going to add them eventually to `.github/workflows/periodic_ci.yml`, but a new one is even better.

File: .github/workflows/vpm_ci.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+name: VPM CI
+
+on:
+  push:
+    paths:
+      - '**/vpm_ci.yml'
+      - 'cmd/tools/vpm.yml'
```
Comment (spytheman) [inline .github/workflows/vpm_ci.yml:7]: hm, there is no `cmd/tools/vpm.yml` file?

Comment (spytheman) [reply .github/workflows/vpm_ci.yml:7]: Do you mean this instead?
```suggestion
      - 'cmd/tools/vpm/**'
```

File: .github/workflows/vpm_ci.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        cc: [tcc, gcc, clang]
+      fail-fast: false
+    env:
+      VFLAGS: -cc ${{ matrix.cc }} -d network
+    steps:
+      - uses: actions/checkout@v4
+      - name: Build V
+        run: |
+          make -j4
+          [[ ${{ matrix.cc }} != "tcc" ]] && ./v -cc ${{ matrix.cc }} -cg -cstrict -o v cmd/v
```
Comment (spytheman) [inline .github/workflows/vpm_ci.yml:31]: there is no need for that, `make` is enough

File: .github/workflows/vpm_ci.yml (hunk 1 above)
Comment (ttytm) [reply .github/workflows/vpm_ci.yml:7]: oops, thank you!

File: .github/workflows/vpm_ci.yml (hunk 2 above)
Comment (ttytm) [reply .github/workflows/vpm_ci.yml]: Oh right, I would add a setup step then to build V once per os instead of re-build V for every matrix configuration.

**Review by spytheman — APPROVED** [review]

**Comment by spytheman** [comment]
Excellent work.

**Pull request merged**